pkg/syncer/syncer.go

```diff
@@ -249,71 +233,52 @@ type holder struct {
 }
 
 func (c *Controller) AddToQueue(gvr schema.GroupVersionResource, obj interface{}) {
-	c.queue.AddRateLimited(holder{gvr: gvr, obj: obj})
+	c.queue.Add(holder{gvr: gvr, obj: obj})
 }
 
 // Start starts N worker processes processing work items.
 func (c *Controller) Start(ctx context.Context, numThreads int) {
-	c.ctx, c.cancelFn = context.WithCancel(ctx)
+	defer runtime.HandleCrash()
+	defer c.queue.ShutDown()
 
-	c.fromInformers.Start(c.ctx.Done())
-	c.fromInformers.WaitForCacheSync(c.ctx.Done())
+	c.fromInformers.Start(ctx.Done())
+	c.fromInformers.WaitForCacheSync(ctx.Done())
 
 	for i := 0; i < numThreads; i++ {
-		go c.startWorker(c.ctx)
+		go wait.UntilWithContext(ctx, c.startWorker, time.Second)
```

Comment:
What's the motivation for this change?

Reply:
This is consistent with standard controller design. It restarts any crashed/stopped worker goroutines once per second if needed (but ideally they don't crash/stop).